Eth-Crowdfund/eth_crowdfund_be
# Api auth

## What's this PR do?
* Generates a starting point if we want to implement api_key authentication app wide
* Creates an ApiKey model & table. columns = id, key, email
* Creates endpoint for generating an API key
* Creates an authenticated endpoint for viewing all api-keys given correct api-key in header
## Where should the reviewer start? 
* ApiKeyView.py
## How should this be manually tested?
* python manage.py db upgrade
  * check database with `psql`
  * make sure you're in the pipenv shell
  * if no api_key table then you may have to run `python manage.py db downgrade`
* postman
* generate a new key: `POST http://localhost:3000/api/v1/api_key/` `body: {"email":"anything@email.com"}`
  * can generate as many as you want. email is required but not unique
* save that key, `GET http://localhost:3000/api/v1/api_key/` `header: api-key: <saved key>`
  * success: show all keys
  * error if no key is provided 
  * error if the wrong key is provided
## Any background context you want to provide?
* good feature that shouldn't affect the FE while improving our security
## What are the relevant tickets?
* #64, but I wouldn't say it's closed
